src/Zicht/Bundle/UrlBundle/Aliasing/Doctrine/CreateAliasSubscriber.php

```diff
@@ -20,21 +20,38 @@ class CreateAliasSubscriber extends BaseSubscriber
     private $isHandling = false;
 
     /**
+     * @var array
+     */
+    private $records = [];
+
+    /**
      * @return array
      */
     public function getSubscribedEvents()
     {
         if ($this->enabled) {
-            return array(
+            return [
                 Events::postPersist,
                 Events::postUpdate,
                 Events::postFlush,
-            );
+            ];
         } else {
-            return array();
+            return [];
         }
     }
 
+    protected function addRecord($entity, array $action = [])
+    {
+        if ($entity instanceof $this->className) {
+            if (false !== ($index = array_search($entity, array_column($this->records, 0), true))) {
```

Comment:
los van dat het nu meer in de array stop en die logica niet meer werkt, kan ik me niet zo goed een voorstelling maken waarom dat sneller zou moeten zijn (en of je dat zou merken). Ken de 
internals ook niet zo goed maar zover ik weet moet hij zowel bij een key index of een searh bij allebij door de array lopen om te kijken of iets bestaat. Maar zou zeggen testen is weten.